core/src/test/java/hudson/PluginManagerTest.java

```diff
@@ -68,4 +84,77 @@ public void parseInvalidRequestedPlugins() throws Exception {
             assertThat(ex.getCause().getMessage(), containsString("Refusing to resolve entity with publicId(null) and systemId (file:///)"));
         }
     }
+    
+    @Test
+    public void shouldProperlyParseManifestFromJar() throws IOException {
+        File jar = createHpiWithManifest();
+        final Manifest manifest = PluginManager.parsePluginManifest(jar.toURI().toURL());
+        
+        assertThat("manifest should have been read from the sample", manifest, notNullValue());
+        assertAttribute(manifest, "Created-By", "Apache Maven");
```

Comment:
If you're testing that this is correctly parsing the manifest, maybe include a test for the line wrapped attributes?